mcs/class/System.ServiceModel/System.ServiceModel.Channels.Http/HttpReplyChannel.cs

```diff
@@ -198,14 +198,22 @@ protected Message CreatePostMessage (HttpContextInfo ctxi)
 				return null;
 			}
 
-			if (!Encoder.IsContentTypeSupported (ctxi.Request.ContentType)) {
+			if (ctxi.Request.ContentType == null || !Encoder.IsContentTypeSupported (ctxi.Request.ContentType)) {
 				ctxi.Response.StatusCode = (int) HttpStatusCode.UnsupportedMediaType;
 				ctxi.Response.StatusDescription = String.Format (
 						"Expected content-type '{0}' but got '{1}'", Encoder.ContentType, ctxi.Request.ContentType);
 				ctxi.Close ();
 
 				return null;
 			}
+			
+			if (ctxi.Request.ContentLength64 == 0) {
```

Comment:
hmm the code in the `#if false // FIXME` makes it sound like a zero-length `ctxi.Request.ContentLength64` is possible in non-buffered mode. does that make sense?

Reply:
I don't use non-buffered mode.
My main problem is that receiving a POST without message body in a WCF Self Hosted application the code is crashing the application. 
As I can see any exception in the "TryReceiveRequest" method will crash the application and not respond the correct HTTP response as Microsoft WCF does.

Reply:
Does not make sense to get all exception on "System.ServiceModel.Channels.ReplyChannelBase.BeginTryReceiveRequest" and return error in the context? This would avoid to propagate exception and crash application. At least in SelfHost application...

Reply:
@atsushieno 
Do you have ideia about the @akoeplinger concerns?

I did this change because my application needs to use the HTTP methods. I used WCF for a REST API and right now I just need to have this pull accepted to release my application.

Can you help me guys?
Tks

Reply:
sorry but too old code and /me being inactive in mono codebase, let's not count on me.